pennylane/ops/qubit.py

```diff
@@ -1580,6 +1634,78 @@ def decomposition(D, wires):
         return [QubitUnitary(np.diag(D), wires=wires)]
 
 
+class QFT(Operation):
+    r"""QFT(wires)
+    Apply a quantum Fourier transform (QFT).
+
+    For the :math:`N`-qubit computational basis state :math:`|m\rangle`, the QFT performs the
+    transformation
+
+    .. math::
+
+        |m\rangle \rightarrow \frac{1}{\sqrt{2^{N}}}\sum_{n=0}^{2^{N} - 1}\omega_{N}^{mn} |n\rangle,
+
+    where :math:`\omega_{N} = e^{\frac{2 \pi i}{2^{N}}}` is the :math:`2^{N}`-th root of unity.
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 0
+    * Gradient recipe: None
+
+    Args:
+        wires (int or Iterable[Number, str]]): the wire(s) the operation acts on
+    """
+    num_params = 0
+    num_wires = AnyWires
+    par_domain = None
+    grad_method = None
+
+    @property
+    def matrix(self):
+        # Redefine the property here to allow for a custom _matrix signature
+        mat = self._matrix(len(self.wires))
+        if self.inverse:
+            mat = mat.conj()
+        return mat
+
+    @classmethod
+    @functools.lru_cache()
+    def _matrix(cls, num_wires):
+        dimension = 2 ** num_wires
+
+        mat = np.zeros((dimension, dimension), dtype=np.complex128)
+        omega = np.exp(2 * np.pi * 1j / dimension)
+
+        for m in range(dimension):
+            for n in range(dimension):
+                mat[m, n] = omega ** (m * n)
+
+        return mat / np.sqrt(dimension)
+
+    @staticmethod
+    def decomposition(wires):
+        num_wires = len(wires)
```

Comment:
When running the QFT, is there a way to make it print out the full decomposition when using `draw`? Currently it draws this:
```
 0: ──╭QFT──╭┤ State 
 1: ──├QFT──├┤ State 
 2: ──╰QFT──╰┤ State 
```
but for the sake of testing it'd be convenient to see the innards.

Reply:
This could be the moment that our new `qml.draw()` shines!

Since it is not dependent on the QNode state, we can add some logic to `qml.draw()` to always decompose the QFT operation (or maybe any operation that has a `draw_decomp` flag or something).

Reply:
It'd be great to have it at the level of the draw function itself - 
```
qml.draw(circuit, draw_decomp=True)
```

Reply:
Or even
```python
qml.draw(qnode, decompose=[“QFT”, “Rot”])(params)
```